src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,160 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+        if (head == null) {
+            addFirst(value);
+        } else if (size == index) {
+            addLast(value);
+        } else {
+            Node<T> currentIndexNode = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(currentIndexNode.prev, value, currentIndexNode);
+            if (currentIndexNode.prev != null) {
+                currentIndexNode.prev.next = newNode;
+            } else {
+                head = newNode;
+            }
+            currentIndexNode.prev = newNode;
+        }
+        size++;
+    }
+
+    private void addFirst(T value) {
+        Node<T> nodeToAdd = new Node<>(null, value, null);
+        head = nodeToAdd;
+        tail = nodeToAdd;
+    }
```

Comment:
The addFirst method does not increment the size of the list. This could lead to inconsistencies when checking the size of the list after adding the first element. Consider adding 'size++' in this method.